models/staging/stg_soroban_contract_code.sql

```diff
@@ -0,0 +1,191 @@
+{{ config(
+    materialized = 'view',
+    tags = ["soroban_analytics"],
+    enabled = true,
+    tests = {
+        'unique': {
+            'column_name': 'ledger_key_hash'
+        },
+        'not_null': {
+            'column_name': 'ledger_key_hash'
+        }
+    }
+) }}
+
+{% set target_table_query %}
+    SELECT COUNT(*) AS tgt_row_count FROM {{ ref('contract_details_hist') }}
+{% endset %}
+
+{% set results = run_query(target_table_query) %}
+
+{% if execute %}
+    {% set is_empty_target_table = (results.columns[0].values()[0] == 0) %}
+{% else %}
+    {% set is_empty_target_table = false %}
+{% endif %}
```

Comment:
What are you trying to accomplish with this logic?

I think dbt automatically creates the table on the first run and I _think_ that it creates this table with the --full-refresh option